lib/receptor_controller/client/response_worker.rb

```diff
@@ -39,13 +40,23 @@ def initialize(config, logger)
 
     # Start listening on Kafka
     def start
+      init_lock, init_wait = Mutex.new, ConditionVariable.new
+
       lock.synchronize do
         return if started.value
 
+        default_messaging_opts # Thread-safe init
+        init_notifications(init_lock, init_wait)
+
         started.value         = true
         workers[:maintenance] = Thread.new { check_timeouts while started.value }
         workers[:listener]    = Thread.new { listen while started.value }
       end
+
+      # Wait for kafka initialization
+      wait_for_notifications(init_lock, init_wait)
```

Comment:
I know it's a stretch - but could we maybe have a timeout for this? If the message hasn't been received back for like 15-20 seconds or something it could really indicate we have issues.

Reply:
done